cmake/OpenCVDetectPython.cmake

```diff
@@ -295,7 +295,12 @@ find_python("${OPENCV_PYTHON3_VERSION}" "${MIN_VER_PYTHON3}" PYTHON3_LIBRARY PYT
 OCV_OPTION(PYTHON3_LIMITED_API "Build with Python Limited API (not available with numpy >=1.15 <1.17)" NO
            VISIBLE_IF PYTHON3_NUMPY_VERSION VERSION_LESS "1.15" OR NOT PYTHON3_NUMPY_VERSION VERSION_LESS "1.17")
 if(PYTHON3_LIMITED_API)
-  set(PYTHON3_LIMITED_API_VERSION "0x03060000" CACHE STRING "Minimal Python version for Limited API")
+  # set LIMITED API version that corresponds to current python version, if python3 is too old
+  if(${PYTHON}_VERSION_STRING VERSION_GREATER "3.5.999")
```

Comment:
Perhaps it would be more readable to use negative condition (+avoid docstring duplication):
```.cmake
set(_default_ver "0x03060000")
if(${PYTHON}_VERSION_STRING VERSION_LESS "3.6")
  # fix for older pythons
  set(_default_ver "0x030${PYTHON3_VERSION_MINOR}0000")
endif()
set(PYTHON3_LIMITED_API_VERSION ${_default_ver} CACHE STRING "Minimal Python version for Limited API")
```